qemu/tests/virtual_nic_stress.py

```diff
@@ -0,0 +1,93 @@
+import logging
+
+from virttest import error_context
+from virttest import utils_net
+from virttest import utils_test
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Guest network stress when under memory stress
+    1) Boot a guest with vhost=on
+    2) swapoff in guest
+    3) flood ping from guest to host
+    4) do stress test
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def flood_ping(session, host_ip):
+        """
+        Do flood ping from guest to host
+
+        :param session: session to send flood ping
+        :param host_ip: the IP of the host
+        """
+        flood_minutes = int(params["flood_minutes"])
+        logging.info("Flood ping for %s minutes" % flood_minutes)
+        # TODO: windows guest doesn't have proper flood ping like linux,
+        # will consider how to implement it for windows later
+        if params.get("os_type") == "linux":
+            utils_net.ping(host_ip, flood=True,
+                           session=session, timeout=flood_minutes * 60)
+
+    def load_stress():
+        """
+        Load background IO/CPU/Memory stress in guest
+
+        """
+        error_context.context("launch stress app in guest", logging.info)
+        args = (test, params, env, params["stress_test"])
+        bg_test = utils_test.BackgroundTest(
+            utils_test.run_virt_sub_test, args)
+        bg_test.start()
+        if not utils_misc.wait_for(bg_test.is_alive, first=10,
+                                   step=3, timeout=100):
+            test.fail("background test start failed")
+
+    def unload_stress(session):
+        """
+        Stop stress app
+
+        :param session: guest session
+        """
+        def _unload_stress():
+            cmd = params.get("stop_cmd")
+            session.sendline(cmd)
+            return True
+
+        error_context.context("stop stress app in guest", logging.info)
+        utils_misc.wait_for(_unload_stress, first=2.0,
+                            text="wait stress app quit",
+                            step=1.0, timeout=120)
```

Comment:
If you only `sendline()` it will return immediately without waiting for the command return, so actually `wait_for()` here is unnecessary.